pandas/core/index.py

```diff
@@ -2119,11 +2124,27 @@ def _maybe_cast_slice_bound(self, label, side):
         label : object
         side : {'left', 'right'}
 
+        Returns
+        -------
+        label :  object
+
         Notes
         -----
         Value of `side` parameter should be validated in caller.
 
         """
+
+        # pass thru float indexers if we have a numeric type index
```

Comment:
This logic here seems OK, but a little confusing in context. To me it would make more sense to put it on the subclass `_maybe_cast_slice_bound` methods, even if that means a little more duplication. Otherwise, things will break, e.g., as soon as we add `IntervalIndex`.